src/containers/pages/FocusInvestigation/map/active/index.tsx

```diff
@@ -153,319 +152,344 @@ export const defaultMapSingleFIProps: MapSingleFIProps = {
   supersetService: supersetFetch,
 };
 
-/** Map View for Single Active Focus Investigation */
-class SingleActiveFIMap extends React.Component<
-  RouteComponentProps<RouteParams> & MapSingleFIProps,
-  FlexObject
-> {
-  public static defaultProps = defaultMapSingleFIProps;
-  constructor(props: RouteComponentProps<RouteParams> & MapSingleFIProps) {
-    super(props);
+const fetchData = async (
+  fetchGoalsActionCreator: typeof fetchGoals,
+  fetchJurisdictionsActionCreator: typeof fetchJurisdictions,
+  fetchPlansActionCreator: typeof fetchPlans,
+  fetchStructuresActionCreator: typeof setStructures,
+  fetchTasksActionCreator: typeof fetchTasks,
+  plan: Plan,
+  supersetService: typeof supersetFetch
+): Promise<void> => {
+  if (plan && plan.plan_id) {
+    /** define superset filter params for jurisdictions */
+    const jurisdictionsParams = superset.getFormData(SUPERSET_MAX_RECORDS, [
+      { comparator: plan.jurisdiction_id, operator: '==', subject: JURISDICTION_ID },
+    ]);
+    await supersetService(SUPERSET_JURISDICTIONS_SLICE, jurisdictionsParams)
+      .then((result: Jurisdiction[]) => fetchJurisdictionsActionCreator(result))
+      .catch(error => {
+        throw error;
+      });
+    /** define superset params for filtering by plan_id */
+    const supersetParams = superset.getFormData(SUPERSET_MAX_RECORDS, [
+      { comparator: plan.plan_id, operator: '==', subject: PLAN_ID },
+    ]);
+    /** define superset params for goals */
+    const goalsParams = superset.getFormData(
+      SUPERSET_MAX_RECORDS,
+      [{ comparator: plan.plan_id, operator: '==', subject: PLAN_ID }],
+      { action_prefix: true }
+    );
+    /** Implement Ad hoc Queries since jurisdictions have no plan_id */
+    await supersetService(SUPERSET_STRUCTURES_SLICE, jurisdictionsParams)
+      .then((structuresResults: Structure[]) => {
+        fetchStructuresActionCreator(structuresResults);
+      })
+      .catch(error => {
+        throw error;
+      });
+    await supersetService(SUPERSET_PLANS_SLICE, jurisdictionsParams)
+      .then((result2: Plan[]) => {
+        fetchPlansActionCreator(result2);
+      })
+      .catch(error => {
+        throw error;
+      });
+    await supersetService(SUPERSET_GOALS_SLICE, goalsParams)
+      .then((result3: Goal[]) => {
+        fetchGoalsActionCreator(result3);
+      })
+      .catch(error => {
+        throw error;
+      });
+    await supersetService(SUPERSET_TASKS_SLICE, supersetParams)
+      .then((result4: Task[]) => {
+        fetchTasksActionCreator(result4);
+      })
+      .catch(error => {
+        throw error;
+      });
+  }
+};
```

Comment:
1. Please consider using [displayError](https://github.com/onaio/reveal-frontend/blob/acdc40f222c9a3bf281c5f2e90ffe180ba34f853/src/helpers/errors.ts#L9) instead of throwing.
2. Please also consider that the result inside the `then` might be false-y, in which case an HTTP error occurred (think 4xx or 5xx errors).  In this case, we also need to use `displayError`.  Look at [how we did it here](https://github.com/onaio/reveal-frontend/pull/692/files#diff-79ae51a7daeac08c863a2be206d150cdR58-R63) - and please suggest any improvements to that process in this PR.

Reply:
> 2\. Please also consider that the result inside the `then` might be false-y, in which case an HTTP error occurred (think 4xx or 5xx errors).  In this case, we also need to use `displayError`.  Look at [how we did it here](https://github.com/onaio/reveal-frontend/pull/692/files#diff-79ae51a7daeac08c863a2be206d150cdR58-R63) - and please suggest any improvements to that process in this PR.

I also noticed that we display the raw error message. I think the error returned bycatch is useful for  a developer and can be logged to sentry but not useful to the user. I think display error should display generic error messages e.g **An error occurred while processing your request** etc

Reply:
@kelvin-muchiri you are absolutely right.  Its my bad - I had meant to create an issue to enhance displayError so that it logs the Error to somewhere, and then show to the user a nice friendly error.

Reply:
I have created the issue here https://github.com/onaio/reveal-frontend/issues/729

Reply:
> 1. Please consider using [displayError](https://github.com/onaio/reveal-frontend/blob/acdc40f222c9a3bf281c5f2e90ffe180ba34f853/src/helpers/errors.ts#L9) instead of throwing.

The error was being displayed by the caller https://github.com/onaio/reveal-frontend/blob/561909bce59fddf4dd47336987c6819df92d4dec/src/containers/pages/FocusInvestigation/map/active/index.tsx#L326. Which one is best. Displaying the error in the caller or the callee?

Reply:
@kelvin-muchiri not sure what you mean, could you restate that?

Reply:
> @kelvin-muchiri not sure what you mean, could you restate that?

I was  catching the errors when I call the method `fetchData`

Reply:
@kelvin-muchiri cool, thanks.  In that case I would leave it as it is.  Only thing left to do is deal with cases where within `fetchData` `supersetService` returns something false-y